### ISSUE TITLE
Automatic ping overtime

### ISSUE THREAD

**Issue by diguifi** [denoland/deno#5511]
I was trying out [ws](https://deno.land/std/ws/mod.ts), and created [this](https://github.com/diguifi/deno-chat) app, but I realised that, after sometime "afk" from the chat, if I try to send a message, I get this error:  
_WebSocket is already in CLOSING or closed state_

I did some research I found out that websockets need to ping overtime to keep the conection alive. I also found out that socket.io (Node) does that automatically:
![image](https://user-images.githubusercontent.com/31022286/82128638-ffbfca00-9792-11ea-8306-7bd5e2cff0fe.png)

As you can see on [this](https://github.com/diguifi/deno-chat/commit/6fe54c77f12507be6358dd1f7dc0796cac6bd1f9) commit, I had to implement the ping system "by hand" on my app.

Is there a better solution to this? Is there a way to "enable" auto-ping on the websocket module?

**Comment by bartlomieju** [comment]
CC @keroxp

**Comment by buckle2000** [comment]
socket.io is not WebSocket. Please close this.

**Comment by keroxp** [comment]
>  I found out that websockets need to ping overtime to keep the conection alive

@diguifi Do you talk about browser-side WebSocket? I don't fully understand what happened to you app. As WebSocket spec, there may be no "timeout" in protocol layer.

**Comment by diguifi** [comment]
@keroxp I'm very new to the websocket world, I'm probably missing something.  
What happened is: I tried [std/ws](https://deno.land/std/ws/) to create a simple chat app over websockets, it worked great, but turns out that after a while without sending or receiving any messages over ws, I get disconected from the socket and start to receive this message  
_WebSocket is already in CLOSING or closed state_

When I tried to figure out what happened, by looking at node projects with similar approach, I found out that, when using socket.io (which I know isn't a native implementation of websockets), it keeps automatically pinging the websocket server overtime.
Based on this discovery, I supposed that I needed to do the same in my project in order to not get disconected after a while afk, but I found weird that std/ws didn't have anything like that implemented by default. That's when I opened the issue, requesting an automatic ping system. I don't even know if thats actually the case to fix my problem... (I also closed the issue just in case this is all absolutely wrong).
